framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3789,18 +3790,18 @@ protected void addAliasedDeclAnnotation(
             @FullyQualifiedName String alias,
             @FullyQualifiedName String annotation,
             AnnotationMirror annotationToUse) {
-        Pair<AnnotationMirror, Set<@FullyQualifiedName String>> pair = declAliases.get(annotation);
-        if (pair != null) {
-            if (!AnnotationUtils.areSame(annotationToUse, pair.first)) {
-                throw new BugInCF(
-                        "annotationToUse should be the same: %s %s", pair.first, annotationToUse);
-            }
-        } else {
-            pair = Pair.of(annotationToUse, new HashSet<>());
-            declAliases.put(annotation, pair);
+        Map<@FullyQualifiedName String, AnnotationMirror> mapping = declAliases.get(annotation);
+        if (mapping == null) {
+            mapping = new HashMap<>(1);
+            declAliases.put(annotation, mapping);
+        }
+        AnnotationMirror prev = mapping.put(alias, annotationToUse);
+        // There already was a mapping. Raise an error.
```

Comment:
Let's see whether this is too restrictive. Maybe when using the command-line aliasing feature #456, this will be an issue.